File: common/src/main/java/org/valkyrienskies/clockwork/integration/cc/ComputerAttachmentHandler.java
package org.valkyrienskies.clockwork.integration.cc;

import dan200.computercraft.api.peripheral.IComputerAccess;

import java.util.ArrayList;
import java.util.List;

public class ComputerAttachmentHandler {
    private final List<IComputerAccess> accessList = new ArrayList<>();

    public void attachComputer(IComputerAccess access) {
        this.accessList.add(access);
    }

    public void detachComputer(IComputerAccess access) {
        this.accessList.remove(access);
    }

    public void sendEvent(String name, Object obj) {
        this.accessList.forEach(access -> access.queueEvent(name, obj));
    }
}

File: common/src/main/kotlin/org/valkyrienskies/clockwork/integration/cc/ComputerAttachmentHandler.kt
package org.valkyrienskies.clockwork.integration.cc

import dan200.computercraft.api.peripheral.IComputerAccess

class ComputerAttachmentHandler {
    private val accessList: MutableList<IComputerAccess> = mutableListOf()

    fun attachComputer(access: IComputerAccess) {
        this.accessList.add(access)
    }

    fun detachComputer(access: IComputerAccess) {
        this.accessList.remove(access)
    }

    fun sendEvent(name: String, vararg obj: Any?) {
        this.accessList.forEach {
            it.queueEvent(name, obj)
        }
    }
}